plasmapy/classes/tests/test_Species.py

```diff
@@ -189,3 +189,15 @@ def test_particle_exb_nonuniform_drift():
     #
     # s.test_kinetic_energy()
 """
+
+def test_particle_nonuniform_grid():
+    '''
+        Test the particle stepper when the spatial domain dimensions are not equal
+    '''   
+    x = np.linspace(0,1,10)*u.m
+    y = np.linspace(0,1,20)*u.m
+    z = np.linspace(0,1,30)*u.m
+    
+    plasma = Plasma(x,y,z)
+    
+    Species(plasma, 'e', dt=1e-14*u.s, nt=1e3)
```

Comment:
I'm not sure this will actually check the interpolator without using the `run` method. How about something like this?

    Species(..., nt = 5).run()

That way we don't waste much time on this test - if it's going to glitch out, 5 iterations are plenty enough of time.

Reply:
The Species() constructor does setup the interpolator and the bug was here. I can add a run() method call, does not cost a lot. Btw is also tested in previous existing tests.